Update filetimes.yml to use conda instead of pip

gbrener opened the pull request:
There are a lot of pip packages listed in examples/filetimes.yml. Since its creation, relevant conda packages have been created and updated. This PR updates the filetimes.yml file to reflect mostly conda packges, with two that are still only available through `pip`.

jbednar commented on examples/filetimes.sh:
```diff
@@ -19,8 +14,8 @@ ${timer} python filetimes.py ${1}.parq        dask    census meterswest metersno
-${timer} python filetimes.py ${1}.castra      dask    census meterswest metersnorth race
-#${timer} python filetimes.py ${1}.castra      pandas  census meterswest metersnorth race
+#${timer} python filetimes.py ${1}.castra      dask    census meterswest metersnorth race
+${timer} python filetimes.py ${1}.castra      pandas  census meterswest metersnorth race
```
Why no dask version of the test for castra anymore?

gbrener replied:
I get `AttributeError: module 'dask.dataframe' has no attribute 'from_castra'` from filetimes.py. I'm pushing up a commit shortly that simply comments it out - perhaps that's a better approach.

jbednar commented:
Looks good! Happy to merge after the one small comment is addressed. After merging, can you please update #129 to show the new performance for a similar set of tests as in the last comments in that issue, now that various libraries involved have been updated?  I can then run that same test on my own machine to see if I get similar results, to make sure the hardware is comparable.

gbrener commented:
Sure, no problem @jbednar . With the commit from a couple minutes ago, I also added some helpful debugging output when an extra argument is provided to `filetimes.sh`, or the `--debug` flag is provided directly to `filetimes.py`. I also moved the timing functionality closer to the actual function calls, to make it easier to rule-out unrelated code affecting the results.

gbrener commented:
One issue that's coming up, when executing the following command (note the `dftype='dask'` argument):
```
python -c "import filetimes as ft ; ft.p.base='census' ; ft.p.x='meterswest' ; ft.p.y='metersnorth' ; ft.p.categories=['race']; ft.timed_write('data/tinycensus.csv',dftype='dask')"
```
I get the following exception:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/gbrener/datashader/examples/filetimes.py", line 164, in timed_write
    duration, res = code(df,fname,p)
  File "/Users/gbrener/datashader/examples/filetimes.py", line 123, in <lambda>
    write["parq"]         ["dask"]   = lambda df,filepath,p:  benchmark(dd.io.parquet.to_parquet, (filepath, df)) ## **p.parq_opts
  File "/Users/gbrener/datashader/examples/filetimes.py", line 95, in benchmark
    res = fn(*posargs, **kwargs)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/dataframe/io/parquet.py", line 295, in to_parquet
    out = delayed(writes).compute()
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/base.py", line 95, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/base.py", line 202, in compute
    results = get(dsk, keys, **kwargs)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/threaded.py", line 76, in get
    **kwargs)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/async.py", line 500, in get_async
    raise(remote_exception(res, tb))
dask.async.TypeError: expected list of bytes

Traceback
---------
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/async.py", line 266, in execute_task
    result = _execute_task(task, data)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/async.py", line 247, in _execute_task
    return func(*args2)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/dask/compatibility.py", line 47, in apply
    return func(*args, **kwargs)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/fastparquet/writer.py", line 580, in make_part_file
    rg = make_row_group(f, data, schema, compression=compression)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/fastparquet/writer.py", line 568, in make_row_group
    compression=comp)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/fastparquet/writer.py", line 453, in write_column
    bdata = encode['PLAIN'](pd.Series(data.cat.categories), selement)
  File "/Users/gbrener/miniconda/envs/filetimes/lib/python3.5/site-packages/fastparquet/writer.py", line 248, in encode_plain
    return pack_byte_array(list(out))
  File "fastparquet/speedups.pyx", line 110, in fastparquet.speedups.pack_byte_array (fastparquet/speedups.c:2570)
```

I don't think this is related to the changes in this branch - I suspect it's coming from a change in the newer version of Dask. Looking into it now.

gbrener commented:
Ok, I believe I've fixed the issue. The cause was a combination of two things:
1. The column names in `tinycensus.csv` don't include the index. Pandas and Dask usually detect the index column automatically, so this normally isn't a problem...
2. The `usecols=[...]` option was being passed to `dd.read_csv()` and `df.read_csv()`, which caused the resulting dataframes to ignore the index column and effectively shift each column one position to the left.
Because the columns were misidentified, Dask choked when the rightmost column (a "category" type) was being written out to parq format for values that were actually floating-point numbers.

jbednar commented:
I'd prefer to just remove the index column entirely from the CSV files, as this will make them smaller, make them match other CSV files more closely, and will let us use `usecols`, which is very important for performance for multi-column datasets.
